smithy-linters/src/main/java/software/amazon/smithy/linters/NoninclusiveTermsValidator.java

```diff
@@ -130,70 +135,79 @@ private Collection<ValidationEvent> getValidationEvents(TextInstance instance) {
-                switch (instance.getLocationType()) {
-                    case NAMESPACE:
-                        //Cannot use any warning() overloads because there is no shape associated with the event.
-                        events.add(ValidationEvent.builder()
-                                .sourceLocation(SourceLocation.none())
-                                .id(this.getClass().getSimpleName().replaceFirst("Validator$", ""))
-                                .severity(Severity.WARNING)
-                                .message(formatNonInclusiveTermsValidationMessage(termEntry, matchedText, instance))
-                                .build());
-                        break;
-                    case APPLIED_TRAIT:
-                        events.add(warning(instance.getShape(),
-                                instance.getTrait().getSourceLocation(),
-                                formatNonInclusiveTermsValidationMessage(termEntry, matchedText, instance)));
-                        break;
-                    case SHAPE:
-                    default:
-                        events.add(warning(instance.getShape(),
-                                instance.getShape().getSourceLocation(),
-                                formatNonInclusiveTermsValidationMessage(termEntry, matchedText, instance)));
-                }
+                events.add(constructValidationEvent(instance, termEntry.getValue(), matchedText));
             }
         }
         return events;
     }
 
-    private static String formatNonInclusiveTermsValidationMessage(
-            Map.Entry<String, List<String>> termEntry,
-            String matchedText,
-            TextInstance instance
+    /**
+     * Creates a single ValidationEvent.
+     * @param instance TextInstance where the noninclusive term was found
+     * @param replacements List of suggested replacements for the noninclusive term
+     * @param matchedText the noninclusive term that was used
+     * @return the constructed ValidationEvent
+     */
+    private ValidationEvent constructValidationEvent(
+            final TextInstance instance,
+            final List<String> replacements,
+            final String matchedText
     ) {
-        final List<String> caseCorrectedEntryValue = termEntry.getValue().stream()
-            .map(replacement -> Character.isUpperCase(matchedText.charAt(0))
-                  ? StringUtils.capitalize(replacement)
-                  : StringUtils.uncapitalize(replacement))
-            .collect(Collectors.toList());
-        String replacementAddendum = !termEntry.getValue().isEmpty()
-                ? String.format(" Consider using one of the following terms instead: %s",
-                    ValidationUtils.tickedList(caseCorrectedEntryValue))
-                : "";
+        final String replacementAddendum = getReplacementAddendum(matchedText, replacements);
```

Comment:
Nit, I know that this class in particular deviates from this, but in general we don't use final neither for arguments nor for local variables.

Reply:
removed final from my changes in this class